indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -93,6 +93,34 @@ public static Path distributedClassPath(Path base)
   }
 
   /**
+   * Dose authenticate against a secured hadoop cluster
+   * In case of any bug fix make sure to fix the code at HdfsStorageAuthentication#authenticate as well.
+   *
+   * @param config containing the principal name and keytab path.
+   */
+  public static void authenticate(HadoopDruidIndexerConfig config)
+  {
+    String principal = config.HADOOP_KERBEROS_CONFIG.getPrincipal();
+    String keytab = config.HADOOP_KERBEROS_CONFIG.getKeytab();
+    if (!Strings.isNullOrEmpty(principal) && !Strings.isNullOrEmpty(keytab)) {
+      Configuration conf = new Configuration();
+      UserGroupInformation.setConfiguration(conf);
```

Comment:
I find it surprising no configuration needs to be populated here, I guess it pulls it all from the classpath configs?